src/sys/socket/addr.rs

```diff
@@ -15,12 +17,24 @@ use std::os::unix::ffi::OsStrExt;
  *
  */
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+#[repr(C)]
+pub struct sockaddr_nl {
```

Comment:
If they make sense for the raw struct in our code, they would probably make sense for the corresponding raw struct in libc. Would you create a PR for libc?